src/daft-sql/src/lib.rs

```diff
@@ -261,6 +261,12 @@ mod tests {
                 JoinType::Inner,
                 None,
             )?
+            .select(vec![
```

Comment:
wildcards now always expand to a `.select(...)`, which shouldnt be an issue as the optimizer should result in the same optimized plan

Reply:
How would this interact with conflicting names BTW?

I wonder what the expected behavior should be if we do `SELECT *, (tbl.A + 1) AS A FROM tbl` and want to "override" the column "A"

Reply:
Also we do have some handling of `col("*")` that @Vince7778 implemented, wonder if we could leverage the same machinery: https://github.com/Eventual-Inc/Daft/pull/2629

Reply:
oh yeah! How did I forget about our `col('*')` support. Will update!